src/components/Search.vue

```diff
@@ -0,0 +1,196 @@
+
+<script>
+import { countlock } from '../utils/mixins';
+
+export default {
+  mixins: [countlock],
+  props: {
+    maxQuickResults: {
+      type: Number,
+      default: 6,
+    },
+    showMore: {
+      type: Boolean,
+      default: true,
+    },
+  },
+  inject: ['girderRest'],
+  data() {
+    return {
+      searchText: '',
+      searchMode: 'prefix',
+      searchTypes: ['user', 'folder', 'item'],
+      searchOptionsMenu: false,
+    };
+  },
+  computed: {
+    loading() {
+      return this.locked;
+    },
+    quickResults() {
+      return this.searchResults.slice(0, this.maxQuickResults);
+    },
+    searchParams() {
+      return {
+        q: this.searchText,
+        mode: this.searchMode,
+        types: JSON.stringify(this.searchTypes),
+        // + 1 to determine if total results > maxQuickResults
+        limit: this.maxQuickResults + 1,
+      };
+    },
+  },
+  asyncComputed: {
+    searchResults: {
+      default: [],
+      async get() {
+        let results = [];
+        this.lock();
+        try {
+          if (this.searchText) {
+            const { data } = await this.girderRest.get('resource/search', {
+              params: this.searchParams,
+            });
+            results = [].concat(...this.searchTypes.map(t => data[t]));
+          }
+        } catch (err) {
+          this.$emit('error', err.message || 'Unknown error during search');
+        }
+        this.unlock();
+        return results;
+      },
+    },
+  },
+  methods: {
+    formatUsername(user) {
+      return `${user.firstName} ${user.lastName} (${user.login})`;
+    },
+  },
+};
+</script>
+
+<template lang="pug">
+v-layout.girder-searchbar(row, align-center)
+  v-icon.mdi-24px(color="white") {{ $vuetify.icons.search }}
+  v-menu.grow.mx-3(
+      offset-y, content-class="girder-searchbar-menu", :open-on-click="false",
+      :value="searchText", :nudge-bottom="6")
+    v-text-field(
+        slot="activator", v-model="searchText", light, solo, hide-details, clearable)
```

Comment:
I'm seeing a pointer cursor on the search text field, tracked it down to the following vuetify rule:

```css
.v-menu__activator * {
    cursor: pointer;
}
```

Is it a bad idea to use a `v-text-field` as an activator for a `v-menu`? Is it easy to do this some other way? If not, we can proceed with this and just override the CSS...

Reply:
> is it easy to do this some other way

I originally did this another (messier) way, and @matthewma7 corrected it to this.  I agree with him that it's more readable this way. 

IMO that isn't an appropriate style for vuetify to apply for v-menu activators, and I might open a bug report about it.

Reply:
From the Vuetify docs:

> open-on-click: Designates whether menu should open on activator click

IMO, if `open-on-click` is false, the cursor pointer shouldn't be used, so it's reasonable for us to file a bug report to see what they have to say. I do understand their rationale for this and I think it's a reasonable component for us to use, so I'm fine with overriding the style.